packages/core/src/asset/AssetPromise.ts

```diff
@@ -71,29 +74,44 @@ export class AssetPromise<T> implements PromiseLike<T> {
         if (this._state === PromiseState.Pending) {
           resolve(value);
           this._state = PromiseState.Fulfilled;
-          this._onProgressCallback = undefined;
+          this._onProgressItemsCallback = undefined;
+          this._onProgressDetailsCallback = undefined;
         }
       };
       const onReject = (reason) => {
         if (this._state === PromiseState.Pending) {
           reject(reason);
           this._state = PromiseState.Rejected;
-          this._onProgressCallback = undefined;
+          this._onProgressItemsCallback = undefined;
+          this._onProgressDetailsCallback = undefined;
         }
       };
       const onCancel = (callback) => {
         if (this._state === PromiseState.Pending) {
           this._onCancelHandler = callback;
         }
       };
-      const setProgress = (progress: number) => {
+      const setItemsProgress = (loaded: number, total: number) => {
```

Comment:
setItemsProgress should has `url` better

Reply:
There are some common requirements such as `AssetPromise.all` does not require url